lib/Diagnostics.php

```diff
@@ -47,79 +48,151 @@ class Diagnostics {
 	/** Everything (summary, single queries with their parameters and events) */
 	const LOG_ALL = '4';
 
-	const EVENT_TYPE = 'EVENT';
-	const QUERY_TYPE = 'QUERY';
-	const SUMMARY_TYPE = 'SUMMARY';
-
 	/** @var \OCP\IConfig */
 	private $config;
+
+	/** @var \OCP\IUserSession */
+	private $session;
 	
 	/** @var \OCA\Diagnostics\Log\OwncloudLog */
 	private $diagnosticLogger;
+
+	/** string */
+	private $diagnosticLevel = null;
+
+	/** string */
+	private $diagnosticForUsers = null;
+
+	/** bool */
+	private $debug = null;
 	
 	/**
 	 * @param \OCP\IConfig $config
 	 */
-	public function __construct(IConfig $config) {
+	public function __construct(IConfig $config, IUserSession $session) {
 		$this->config = $config;
+		$this->session = $session;
 		$this->diagnosticLogger = new OwncloudLog($config);
 	}
-
+	
+	/**
+	 * @param string $uids - e.g. "["admin","user1000"]" in JSON format
+	 */
+	public function setDiagnosticForUsers($uids) {
+		$this->config->setAppValue('diagnostics', 'diagnoseUsers', $uids);
+		$this->diagnosticForUsers = json_decode($uids, true);
+	}
+	
+	/**
+	 * @return string[] $uid
+	 */
+	public function getDiagnosedUsers() {
+		if ($this->diagnosticForUsers === null) {
+			$this->diagnosticForUsers = json_decode($this->config->getAppValue('diagnostics', 'diagnoseUsers', "[]"), true);
+		}
+		return $this->diagnosticForUsers;
+	}
+	
+	/**
+	 * @return bool
+	 */
+	public function isDiagnosticActivatedForSession() {
+		if($this->isDebugEnabled() && ($this->getDiagnosticLogLevel() !== self::LOG_NOTHING)) {
+			// If in debug mode, always enabled
+			return true;
+		} else if ($this->getDiagnosticLogLevel() !== self::LOG_NOTHING) {
+			// If diagnostic level is set, lets check if diagnostic is enabled for this user
+			$user = $this->session->getUser();
+			if ($user) {
+				$diagnosedUsers = $this->getDiagnosedUsers();
+				if (in_array($user->getUID(), $diagnosedUsers)) {
+					return true;
```

Comment:
User associated with session needs to be the same as the one from $this->config->getAppValue('diagnostics', 'diagnoseUsers', "[]")